common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1128,14 +1128,36 @@ def get_parent_location(self, locator, **kwargs):
 
         :param locator: BlockUsageLocator restricting search scope
         """
+        def is_root_block(xblock, is_root=False):
```

Comment:
For the first question, how can we measure this? I'd imagine this is fairly fast, since we're not making any database calls.

For the second, not as much? Even if we do want to fix it, fixing it first for new mongo would make a difference for the courses that are actually currently running and are currently under development.